lib/scala/profiling-utils/src/main/java/org/enso/profiling/snapshot/HeapDumpGenerator.java

```diff
@@ -0,0 +1,38 @@
+package org.enso.profiling.snapshot;
+
+import com.sun.management.HotSpotDiagnosticMXBean;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.nio.file.Path;
+import javax.management.MBeanServer;
+
+public final class HeapDumpGenerator {
+
+  private static final String HOT_SPOT_BEAN_NAME = "com.sun.management:type=HotSpotDiagnostic";
```

Comment:
nit: I think we typically use Camel Case for constants in Java. This seems to be Enso's style?